src/components/views/location/LocationButton.tsx

```diff
@@ -75,6 +81,38 @@ export const LocationButton: React.FC<IProps> = (
     </React.Fragment>;
 };
 
+const shareLocation = (client: MatrixClient, roomId: string, openMenu: () => void) =>
+    (uri: string, ts: number) => {
+        if (!uri) return false;
+        try {
+            const text = textForLocation(uri, ts, null);
+            client.sendMessage(
+                roomId,
+                makeLocationContent(text, uri, ts, null),
+            );
+        } catch (e) {
+            logger.error("We couldn’t send your location", e);
+            Modal.createTrackedDialog(
+                'We couldn’t send your location',
+                '',
+                QuestionDialog,
+                {
+                    title: _t("We couldn’t send your location"),
+                    description: _t(
+                        "Element could not send your location. Please try again later."),
+                    button: _t('Try again'),
+                    cancelButton: _t('Cancel'),
+                    onFinished: (tryAgain: boolean) => {
+                        if (tryAgain) {
+                            openMenu();
+                        }
+                    },
+                },
+            );
+        }
+        return true;
+    };
```

Comment:
floating braces are a sign of poor indentation elsewhere imo

Reply:
How would you improve this section?  Nothing jumps out at me.